docs/guides/features/command.md

```diff
@@ -164,28 +164,61 @@ It also gives you full control over the way your error type gets serialized. In
 
 ## Async Commands
 
+Asynchronous functions are benefical in Tauri to perform heavy work in a manner that doesn't result in UI freezes or slowdowns.
+
 :::note
 
 Async commands are executed on a separate thread using [`async_runtime::spawn`].
 Commands without the _async_ keyword are executed on the main thread unless defined with _#[tauri::command(async)]_.
 
 :::
 
-If your command needs to run asynchronously, simply declare it as `async`:
+
+:::note
+
+You need to be careful when creating asynchronous functions using Tauri. Currently, you cannot include types that are borrowed, such as those with the & symbol, in the constructor of an asynchronous function. Some common examples of types like this are `&str` and `State<'_, Data>`.
+
+:::
+
+
+If your command needs to run asynchronously, simply declare it as `async`, and resolve any issues with borrowed types.
+
+When working with borrowed types, there are two main options.
```

Comment:
```suggestion
If your command needs to run asynchronously, simply declare it as `async`.

When working with borrowed types, you have to make additional changes. These are your two main options:
```

Not quite happy with my suggestion but i hope you're getting the idea.

Reply:
I think the suggestion is great, makes it more fluid. :D

Reply:
Committed!